packages/generator/src/complex-type/file.ts

```diff
@@ -1,20 +1,21 @@
 /* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
 
 import { SourceFileStructure, StructureKind } from 'ts-morph';
-import { VdmComplexType } from '../vdm-types';
+import { VdmComplexType, VdmServiceMetadata } from '../vdm-types';
 import { builderFunction } from './builder-function';
 import { fieldTypeClass } from './field-type-class';
 import { importDeclarations } from './imports';
 import { complexTypeInterface } from './interface';
 import { complexTypeNamespace } from './namespace';
 
 export function complexTypeSourceFile(
-  complexType: VdmComplexType
+  complexType: VdmComplexType,
+  service: VdmServiceMetadata
 ): SourceFileStructure {
   return {
     kind: StructureKind.SourceFile,
     statements: [
-      ...importDeclarations(complexType),
+      ...importDeclarations(complexType, service),
```

Comment:
If you just use the odata version also just pass the odata version

Reply:
Yeah, I thought of that, but found the other way a little more consistent with the rest. Anyways, no big deal, I changed it.

Reply:
thanks. I try to keep interfaces as tight as possible. Then you can already guess ok it will have a v2/v4 switch.